tests/test_pds4_create_xml_index_2.py

```diff
@@ -0,0 +1,232 @@
+from lxml import etree
```

Comment:
`test_pds4_create_xml_index.py`` appears to be "black box" tests - where we run the whole program without understanding how it works internally. `test_pds4_create_xml_index_2.py` appears to be "white box" tests, where we are calling individual functions and doing unit tests on them. I would rename the files to `test_pds4_create_xml_index_whitebox.py` etc.